extra/mcs_backup_manager.sh

```diff
@@ -2763,25 +2773,98 @@ validation_prechecks_for_dbrm_backup() {
     # Confirm storage not empty
     if [ -z "$storage" ]; then printf "[!] Empty storage: \ncheck: grep -m 1 \"^service = \" \$STORAGEMANGER_CNF | awk '{print \$3}' \n\n"; fi;
 
-    # Check backup location exists
+    # Check backup directory exists
     if [ ! -d $backup_location ]; then 
-        echo "[!] \$backup_location: Expected directory of dbrm_backups"
+        printf "[!] \$backup_location: Path of backups does Not exist\n"
+        printf "Path: $backup_location\n\n"
+        exit 2;
+    fi
+
+    # Check specific backup exists 
+    backup_folder_to_restore_dbrms=$backup_folder_to_restore
+    if [ $storage == "S3" ]; then
+        backup_folder_to_restore_dbrms="${backup_folder_to_restore}/dbrms"
+    fi
+
+    if [ ! -d "${backup_location}/${backup_folder_to_restore_dbrms}" ]; then 
+        printf "[!] \$backup_folder_to_restore: Path of backup to restore does Not exist\n"
+        printf "Path: ${backup_location}/${backup_folder_to_restore_dbrms}\n\n"
         exit 2;
+    else   
+        echo " - Backup directory exists"
+        if [ "$(ls -A ${backup_location}/${backup_folder_to_restore_dbrms})" ]; then
+
+            expected_files=(
+                "BRM_saves_current"
+                "BRM_saves_em"
+                "BRM_saves_journal"
+                "BRM_saves_vbbm"
+                "BRM_saves_vss"
+                "BRM_savesA_em"
+                "BRM_savesA_vbbm"
+                "BRM_savesA_vss"
+                "BRM_savesB_em"
+                "BRM_savesB_vbbm"
+                "BRM_savesB_vss"
+                "oidbitmap"
+                "SMTxnID"
+            )
+            
+            # Check if all expected files exist
+            for file in "${expected_files[@]}"; do
+                if [ ! -f "${backup_location}/${backup_folder_to_restore_dbrms}/${file}" ]; then
+                    printf "[!] File not found: ${file} in the DBRM backup directory\n"
+                    printf "Path: ${backup_location}/${backup_folder_to_restore_dbrms}/${file}\n\n"
+                    exit 2;
+                fi
+            done
+
+            printf " - Backup contains all files\n"
+
+        else
+            printf "[!] No files found in the DBRM backup directory\n"
+            printf "Path: ${backup_location}/${backup_folder_to_restore_dbrms}\n\n"
+            exit 2;
+        fi
     fi
 
     # Confirm bucket connection
     if [ "$storage" == "S3" ]; then
         if ! testS3Connection 1>/dev/null 2>/dev/null; then
             printf "\n[!] Failed testS3Connection\n\n"
             exit 1;
+        else
+            echo " - S3 Connection works"
         fi
     fi;
 
+    # Download cs_package_manager.sh if not exists
+    if [ ! -f "cs_package_manager.sh" ]; then
+        wget https://raw.githubusercontent.com/mariadb-corporation/mariadb-columnstore-engine/develop/extra/cs_package_manager.sh; chmod +x cs_package_manager.sh;
```

Comment:
cs_package_manager.sh is in the engine repo, but not installed/packaged.
I guess since cmapi now bundles mcs_backup_manager.sh, it should also bundle cs_package_manager.sh.
Yes both scripts use each other for some functions